config/functions

```diff
@@ -2,23 +2,21 @@ setup_toolchain() {
   TARGET_AR_NM_RANLIB_PREFIX=""
 
   if [ "$LTO_SUPPORT" = "yes" ]; then
+    # static libs with lto info require gcc-ar/gcc-nm/gcc-ranlib instead of
+    # default binutils versions
+    TARGET_AR_NM_RANLIB_PREFIX="gcc-"
     if flag_enabled "lto-parallel" "no"; then
       TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
       TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
       TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_PARALLEL"
-      # static libs with lto info require gcc-ar/gcc-nm/gcc-ranlib instead of
-      # default binutils versions
-      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
     elif flag_enabled "lto-fat" "no"; then
       TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_FAT"
       TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_FAT"
       TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_NO_PARALLEL"
-      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
     elif flag_enabled "lto" "no"; then
       TARGET_CFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
       TARGET_CXXFLAGS+=" $FLAGS_OPTIM_LTO_NO_PARALLEL $FLAGS_OPTIM_LTO_NO_FAT"
       TARGET_LDFLAGS+=" $LDFLAGS_OPTIM_LTO_COMMON $FLAGS_OPTIM_LTO_NO_PARALLEL"
-      TARGET_AR_NM_RANLIB_PREFIX="gcc-"
```

Comment:
Some packages will execute the (missing) `else` in this `if..elif..elif` statement, eg. `linux` because `$LTO_SUPPORT` is enabled, but none of the `lto-parallel`/`lto-fat`/`lto` checks evaluate to true. In this case, `TARGET_AR_NM_RANLIB_PREFIX` will be set to `gcc-`, without any LTO compiler/linker flags being added.

Currently, `TARGET_AR_NM_RANLIB_PREFIX` would be left unset for `linux`, so moving this code around is introducing new behaviour which may be unintentional - can you confirm if this is intentional? If not, probably best to leave it as it is, and just include the patch?

Reply:
I tested just the patch on 16.04 (without any change to `config/functions`) and that failed, so your change looks correct, in which case it could be simplified a little further with:
```
-  TARGET_AR_NM_RANLIB_PREFIX=""
-
   if [ "$LTO_SUPPORT" = "yes" ]; then
   ...
+  else
+    TARGET_AR_NM_RANLIB_PREFIX=""
+  fi
```

Reply:
Done.